src/invoker.js

```diff
@@ -127,6 +130,8 @@ class OpenWhiskInvoker {
             this.sourceDir = this.sourcePath;
             this.sourceFile = "";
         }
+        // sourceRoot is cwd unless a build path has been defined in which case it is buildPathRoot or buildPath directory
+        this.sourceRoot = this.hasBuildPath && (this.buildPathRoot || this.sourceDir) || this.cwd
```

Comment:
@alexkli thanks for the review. :)

- IIUC in #23 we discussed that if buildPath is defined we keep the old way a.k.a parent folder if buildPath is a file or buildPath if buidPath is a dir

- Maybe I am missing smthg but as per the condition `this.sourceDir` should be reachable if `--build-path` is defined (`this.hasBuildPath===true`) and there is no `--build-path-root` option set (`this.buildPathRoot===false`).

Reply:
but yeah I guess I could get rid of requiring the buildPath option if buildPathRoot is set, will check tmw

Reply:
Ah, I thought `hasBuildPath` is based off of `buildPathRoot`.

I think this expression could be simplified to
```
= this.buildPath && (this.buildPathRoot || this.sourceDir) || this.cwd
```

Anyway, it's getting hard to follow, let me cover this through tests.

Reply:
ok let me try to simplify the logic a bit

Reply:
@alexkli I think it's ok to use `this.hasBuildPath` because there is no `this.buildPath`
because of constructor `this.sourcePath = options.buildPath || options.sourcePath;`.

We can of course change this logic and expose both`buildPath` and `sourcePath` as two different entities to the developer of the kind wrapper, but as only buildPath is mounted into the container I don't think it's a good idea

I agree that things are not very clear but not sure how to simplify best
for reference and to recap, as of now we need:
`sourcePath/Dir` as watch path for live reload
`cwd` as mount path if buildPath is not defined
`buildPath/Dir` as mount path if buildRootPath is not defined
`buidRootPath` as mount path if buildPath is defined